src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -34,6 +36,71 @@ class WorkflowDataConditionGroupType(StrEnum):
     WORKFLOW_TRIGGER = "workflow_trigger"
 
 
+def get_actions_to_workflows(actions_to_fire: list[Action]) -> dict[int, Workflow]:
+    action_ids = {action.id for action in actions_to_fire}
+    workflow_actions = (
+        WorkflowDataConditionGroup.objects.select_related("workflow", "condition_group")
+        .filter(condition_group__dataconditiongroupaction__action_id__in=action_ids)
+        .values_list("condition_group__dataconditiongroupaction__action_id", "workflow_id")
+    )
+
+    action_to_workflows: dict[int, int] = {
+        action_id: workflow_id for action_id, workflow_id in workflow_actions
+    }
+    workflow_ids = set(action_to_workflows.values())
+
+    workflows = Workflow.objects.filter(id__in=workflow_ids)
+    workflow_ids_to_workflows = {workflow.id: workflow for workflow in workflows}
+
+    # TODO: account for actions being attached to multiple workflows?
+    # should you even send multiple notifications then? :think:
```

Comment:
no, we should de-dupe the actions and only trigger them once. there was some stuff around that in `trigger_actions` that i turned the action list for all the workflows into a set.